financial-services-accelerator/components/org.wso2.financial.services.accelerator.identity.extensions/src/main/java/org/wso2/financial/services/accelerator/identity/extensions/util/IdentityCommonUtils.java

```diff
@@ -700,4 +710,117 @@ public static List<RequestedClaim> getRequestedClaims(String token) throws Ident
         return requestedClaims;
     }
 
+    /**
+     * Get consent id from the request object or common auth id.
+     *
+     * @param oauthAuthzMsgCtx OAuth authorization message context
+     * @return Consent ID
+     * @throws JsonProcessingException
+     * @throws ConsentManagementException
+     */
+    public static String getConsentId(OAuthAuthzReqMessageContext oauthAuthzMsgCtx) throws JsonProcessingException,
+            ConsentManagementException {
+
+        FinancialServicesConfigParser configParser = FinancialServicesConfigParser.getInstance();
+        if (!configParser.isPreInitiatedConsent()) {
+            String commonAuthId = getCommonAuthId(oauthAuthzMsgCtx);
+            return getConsentIdFromCommonAuthId(commonAuthId);
+        }
+
+        if ("requestObject".equals(configParser.getAuthFlowConsentIdSource())) {
+            return getConsentIdFromRequestObject(oauthAuthzMsgCtx.getAuthorizationReqDTO());
+        }
+
+        if ("requestParam".equals(configParser.getAuthFlowConsentIdSource())) {
+            return getConsentIdFromRequestParam(oauthAuthzMsgCtx.getAuthorizationReqDTO());
+        }
+
+        return null;
+    }
+
+    private static String getConsentIdFromCommonAuthId(String commonAuthId) throws ConsentManagementException {
+
+        return new ConsentCoreServiceImpl()
+                .getConsentIdByConsentAttributeNameAndValue(IdentityCommonConstants.COMMON_AUTH_ID, commonAuthId)
+                .get(0);
```

Comment:
Is the null pointer handled here?

Reply:
Fixed with 533d2b6e24e4c38d24460cd3042132849e579b08